agent/jvm/src/main/java/org/jolokia/jvmagent/security/AllAuthenticator.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jolokia.jvmagent.security;
+
+import com.sun.net.httpserver.Authenticator;
+import com.sun.net.httpserver.HttpExchange;
+
+import java.util.ArrayList;
+
+/**
+ * Authenticator that succeeds if all of the Authenticators it delegates to succeed.
+ *
+ * @author roland
+ * @since 26.05.14
+ */
+public class AllAuthenticator extends Authenticator {
```

Comment:
Just wondering whether an AllAuthenticator makes much sense. Also, it is not used currently, so maybe for KISS reasons we simply omit it now and can easily reintroduce it we allow for a more flexible configuration where an AllAuthenticator could be selected.